src/Codec/Xlsx/Parser.hs

```diff
@@ -54,68 +72,70 @@ data WorksheetFile = WorksheetFile { wfName :: Text
-    cur = case parseLBS def file of
-      Left _  -> error "could not read file"
-      Right d -> fromDocument d
-
-    -- The specification says the file should contain either 0 or 1 @sheetViews@
-    -- (4th edition, section 18.3.1.88, p. 1704 and definition CT_Worksheet, p. 3910)
-    sheetViewList = cur $/ element (n"sheetViews") &/ element (n"sheetView") >=> fromCursor
-    sheetViews = case sheetViewList of
-      [] -> Nothing
-      views -> Just views
-
-    commentsMap = getComments ar . relTarget =<< findRelByType commentsType sheetRels
-    commentsType = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/comments"
-
-    sheetRels = getRels ar (wfPath wf)
-
-    -- Likewise, @pageSetup@ also occurs either 0 or 1 times
-    pageSetup = listToMaybe $ cur $/ element (n"pageSetup") >=> fromCursor
-
-    cws = cur $/ element (n"cols") &/ element (n"col") >=> fromCursor
-
-    (rowProps, cells) = collect $ cur $/ element (n"sheetData") &/ element (n"row") >=> parseRow
-    parseRow c = do
-      r <- c $| attribute "r" >=> decimal
-      let ht = if attribute "customHeight" c == ["true"]
-               then listToMaybe $ c $| attribute "ht" >=> rational
-               else Nothing
-      let s = if attribute "s" c /= []
-              then listToMaybe $ c $| attribute "s" >=> decimal
-              else Nothing
-      let rp = if isNothing s && isNothing ht
-               then  Nothing
-               else  Just (RowProps ht s)
-      return (r, rp, c $/ element (n"c") >=> parseCell)
-    parseCell :: Cursor -> [(Int, Int, Cell)]
-    parseCell cell = do
-      ref <- cell $| attribute "r"
-      let
-        s = listToMaybe $ cell $| attribute "s" >=> decimal
-        t = fromMaybe "n" $ listToMaybe $ cell $| attribute "t"
-        d = listToMaybe $ cell $/ element (n"v") &/ content >=> extractCellValue sst t
-        f = listToMaybe $ cell $/ element (n"f") >=> fromCursor
-        (c, r) = T.span (>'9') ref
-        comment = commentsMap >>= lookupComment ref
-      return (int r, col2int c, Cell s d comment f)
-    collect = foldr collectRow (M.empty, M.empty)
-    collectRow (_, Nothing, rowCells) (rowMap, cellMap) =
-      (rowMap, foldr collectCell cellMap rowCells)
-    collectRow (r, Just h, rowCells) (rowMap, cellMap) =
-      (M.insert r h rowMap, foldr collectCell cellMap rowCells)
-    collectCell (x, y, cd) = M.insert (x,y) cd
-
-    merges = cur $/ parseMerges
-    parseMerges :: Cursor -> [Text]
-    parseMerges = element (n"mergeCells") &/ element (n"mergeCell") >=> parseMerge
-    parseMerge c = c $| attribute "ref"
-
-    condFormtattings = M.fromList . map unCfPair  $ cur $/ element (n"conditionalFormatting") >=> fromCursor
+             -> Parser Worksheet
+extractSheet ar sst wf = do
+  let filePath = wfPath wf
+  file <- note (MissingFile filePath) $ Zip.fromEntry <$> Zip.findEntryByPath filePath ar
+  cur <- either (\_ -> throwError $ InvalidFile filePath) (return . fromDocument) $ 
```

Comment:
I don't think that would work, because the monads are different (`Either SomeException` vs. `Either ParseError`). Any suggestions? It's possible I'm missing something here.